### PR TITLE
Allow to define log and data folder and add python3 compatibility

### PR DESCRIPTION
**Pull request by tomamplius**
leave default location in / but allow to set other location for data folder

### CODE REVIEW

**Comment by derneuere** [comment]
Thanks for your pull request. 

I don't understand which problem you want to solve. This will affect the placement within the docker container. You are able to set your folder as you wish by setting the variable "myPhotos"  in the .env. 

What benefits does this bring?

**Comment by tomamplius** [comment]
I have create script to deploy libre photo on debian. Use folder in root / is not a good practice in this case

**Comment by derneuere** [comment]
Ah, I see. Could you add the script for debian to this pull request? This should make it more obvious why it is needed.

**Comment by tomamplius** [comment]
It's **my first working version** of my script is here : https://git.lgy.fr/thomas/install-scripts/src/branch/master/librephotos.sh
It's compatible with the current version of librephoto
I want create new version of my mscript with separed systemd service and others change like create configuration file for environnement variable
But let me propose more pull request with change log and data_models location.
And i want to work on librephotos compatiblity with python 3.9 available in debian 11
After i can add the script 

On next script version with /var/lib/librephoto i will add
export BASE_DATA = /var/lib/librephotos
and change the location in ngnix
~~~
    # Django media
    location /protected_media  {
        internal;
        alias /var/lib/librephoto/protected_media/;      
    }
   # Original Photos
    location /original  {
        internal;
        alias  /var/lib/librephoto/data/;
    }
    # Nextcloud Original Photos
    location /nextcloud_original  {
        internal;
        alias  /var/lib/librephoto/data/nextcloud_media/;
    }

**Comment by derneuere** [comment]
Where does LibrePhotos use encodestring? Is it an usused dependency?

**Comment by tomamplius** [comment]
see api/models/photo.py
Yes in geopy 1.11.0
But 2.1.0 is full compatible

**Comment by tomamplius** [comment]
sorry i'm not work on the good project tomamplius/librephotos
I will work on my repository.
I don't know if you want work to allow  more easily  install on debian
I don't know if you prefer a full pull request or many small pull request

**Comment by tomamplius** [comment]
do you want create a repository like librephotos-docker for linux/debian deployement (librephotos-linux) ?
the goal is store systemd file, configuration, file, and deployement script and maybe deb file creation script.

**Comment by derneuere** [comment]
I think I would prefer a complete pull request. It makes it easier to test for me, because I have to check all the functions anyway. I will probably test if the docker build still works even if it was used before. 

I created a repository, where you can put the debian scripts and configurations: https://github.com/LibrePhotos/librephotos-linux

**Comment by tomamplius** [comment]
Done for me. 
All my work is in the other repository now. You can validate with no regression test
tkx

File: ownphotos/settings.py [hunk 1]
```diff
@@ -228,11 +240,24 @@
...
 
+# Allow to define data folder like /var/lib/librephotos
+
+BASE_DATA = ''
+if 'BASE_DATA' in os.environ:
+	BASE_DATA = os.environ['BASE_DATA']
+
+BASE_LOGS='/logs/'
+if 'BASE_LOGS' in os.environ:
+	BASE_LOGS = os.environ['BASE_LOGS']
```
Comment (gawel) [inline ownphotos/settings.py]: you can use a one liner here:

`BASE_LOGS = os.environ.get('BASE_LOGS', '/logs')`

also making `/var/logs/librephotos` the defaults maybe a good idea. this is the standard in many (all ?) linux distribution.

Comment (tomamplius) [reply ownphotos/settings.py]: done
Agree about /var/log/librephotos (without s on log)
But it's need change on docker deployement

**Comment by gawel** [comment]
I'm currently trying the branch and it seems that exceptions are not logged. Makes it pretty hard to debug... I'm experimenting some 500 and can't find any traceback

**Comment by tomamplius** [comment]
> I'm currently trying the branch and it seems that exceptions are not logged. Makes it pretty hard to debug... I'm experimenting some 500 and can't find any traceback

How you experimenting some 500?
Can you confirm that you use docker?

**Comment by gawel** [comment]
I've added this to settings.py and now errors shows in stdout (journald in my case):
```
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
        },
    },
    'loggers': {
        'django': {
            'handlers': ['console'],
            'level': 'INFO',
        },
    },
}
```

**Comment by tomamplius** [comment]
Ok, It's not link to this pull request? only path are changed.

**Comment by gawel** [comment]
No, not linked to this PR but I'm using this branch. Probably more related to #170

**Comment by gawel** [comment]
The error was caused by a wrong nextcloud url.

The PR works for me. Got my instance up and running.

+1 for the merge

**Comment by tomamplius** [comment]
tkx

File: ownphotos/settings.py [hunk 2]
```diff
@@ -228,11 +240,19 @@
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/1.11/howto/static-files/
 
+# Allow to define data folder like /var/lib/librephotos
+
+BASE_DATA = os.environ.get('BASE_DATA', '')
```
Comment (derneuere) [inline ownphotos/settings.py]: This should be BASE_DATA = os.environ.get('BASE_DATA', '/'), because otherwise it will not find the folder under docker

**Comment by derneuere** [comment]
Thanks again for your pull request. I tested it and it seems to work now after I changed the base path.

**Pull request merged**